collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2478,34 +2450,105 @@ public void denied() {
+                        } else {
+                            resumeFormEntry(formController, warningMsg);
+                        }
                     }
                 }
+            }
 
-                // Register to receive location provider change updates and write them to the audit
-                // log. onStart has already run but the formController was null so try again.
-                if (formController.currentFormAuditsLocation()
-                        && PlayServicesUtil.isGooglePlayServicesAvailable(this)) {
-                    registerReceiver(locationProvidersReceiver, new IntentFilter(LocationManager.PROVIDERS_CHANGED_ACTION));
-                }
+        } else {
+            Timber.e("FormController is null");
+            ToastUtils.showLongToast(R.string.loading_form_failed);
+            finish();
+        }
+    }
 
-                // onStart ran before the form was loaded. Let the viewModel know that the activity
-                // is about to be displayed and configured. Do this before the refresh actually
-                // happens because if audit logging is enabled, the refresh logs a question event
-                // and we want that to show up after initialization events.
-                activityDisplayed();
+    private void showIdentityPromptDialog(FormController formController, boolean startingNewForm, String warningMsg) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        IdentifyUserPromptDialogFragment dialog = IdentifyUserPromptDialogFragment.create(getFormController().getFormTitle());
+        dialog.show(fragmentManager.beginTransaction(), IdentifyUserPromptDialogFragment.TAG);
 
-                refreshCurrentView();
+        IdentityPromptViewModel.Factory factory = new IdentityPromptViewModel.Factory(getFormController().getAuditEventLogger());
+        IdentityPromptViewModel identityPromptViewModel = ViewModelProviders.of(this, factory).get(IdentityPromptViewModel.class);
 
-                if (warningMsg != null) {
-                    ToastUtils.showLongToast(warningMsg);
-                    Timber.w(warningMsg);
+        identityPromptViewModel.isIdentitySet().observe(this, isIdentitySet -> {
+            if (isIdentitySet) {
+                if (startingNewForm) {
+                    startFormEntry(formController, warningMsg);
+                } else {
+                    resumeFormEntry(formController, warningMsg);
                 }
             }
-        } else {
-            Timber.e("FormController is null");
-            ToastUtils.showLongToast(R.string.loading_form_failed);
-            finish();
+        });
+
+        identityPromptViewModel.isFormEntryCancelled().observe(this, isFormEntryCancelled -> {
+            if (isFormEntryCancelled) {
+                onBackPressed();
+            }
+        });
+    }
+
+    private void startFormEntry(FormController formController, String warningMsg) {
```

Comment:
Start and resume are identical?

Reply:
Yeah I actually noticed that while looking into accounting for the last pathway into the form (comments above). This will get fixed.